File: src/slang/parse/Precedence.java
package slang.parse;

public class Precedence {
    public static final int SEQUENCE = 1;
    public static final int ASSIGNMENT = 2;
    public static final int APPLY = 3;
    public static final int CONDITIONAL = 4;
    public static final int SUM = 5;
    public static final int PRODUCT = 6;
    public static final int EXPONENT = 7;
    public static final int PREFIX = 8;
    public static final int POSTFIX = 9;
    public static final int CALL = 10;
}

File: src/slang/parse/Precedence.scala
package slang.parse

object Precedence extends Enumeration {
    type Precedence = Int

    val SEQUENCE = 1
    val CALL = 998
    val PREFIX = 999
    val POSTFIX = 1000
}

sealed trait Associativity
object Associativity {
    case object Left extends Associativity
    case object Right extends Associativity
    case object NonAssoc extends Associativity
}